src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -72,6 +81,26 @@ protected synchronized void load(Path path) throws IOException {
     }
   }
 
+  /**
+   * Attempts to load data from the file on disk into the SlidingWindow
+   *
+   * <p>If this function fails to load the file it will just print an error log and the caller will
+   * be allowed to continue on. It is the caller's responsibility to ensure their data has been loaded
+   * properly into the SlidingWindow.
+   */
+  protected void tryLoad() {
```

Comment:
If we remove the try-catch (let caller handle) and have `if (!enablePersistence)` check in `load()` then do we need this fn? 
same for `tryWrite()`

Reply:
Removed both